pgbouncer/README.md

```diff
@@ -0,0 +1,96 @@
+
+## Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- [Multipass](https://multipass.run/)
+- Docker (for rendering the cloud-init configuration)
+- Git (for cloning the repository)
+
+## Quick Start for New Users
+
+To get started with the sample app, follow these steps:
+
+1. **Clone the Repository**: 
+   ```sh
+   git clone https://github.com/grafana/integration-sample-apps.git
+   cd integration-sample-apps/apache-solr
+   ```
+
+2. **Set Up Default Config**: 
+   Execute `make defaultconfig` to create a template file with default configuration variables. Modify `jinja/variables/cloud-init.yaml` to connect the Grafana agent to an external Prometheus compatible TSDB and/or Loki server.
+
+3. **Render Cloud-init Configuration**: 
+   Run `make render-config` to generate the `cloud-init.yaml` file based on your configuration.
+
+4. **Create and Set Up VMs**: 
+   Use `make run` to start the PgBouncer sample app.
+
+5. **Fetch Prometheus Metrics**: 
+   Fetch metrics from the Prometheus exporter and save them with `make fetch-prometheus-metrics`.
+
+6. **Stop and Clean Up**: 
+   Use `make stop` to clean up the VM and `make clean` to remove temporary files.
+
+## Make Commands
+
+- `make defaultconfig`: Initializes the configuration file with default values for cloud-init templates.
+- `make render-config`: Generates the `cloud-init.yaml` configuration file using the defined variables.
+- `make run`: Creates the PgBouncer sample app.
+- `make load-test`: Generates load on the Solr cluster for testing purposes.
+- `make fetch-prometheus-metrics`: Fetches metrics from the Prometheus exporter and saves them to a local file.
+- `make setup-grafana-agent`: Sets up Grafana Agent on each VM for forwarding metrics and logs.
+- `make clean`: Deletes all created VMs and performs cleanup.
+
+## Default Configuration Variables
+
+- `prom_pass`: Your Prometheus password.
+- `prom_user`: Your Prometheus username.
+- `prom_url`: URL for Prometheus push endpoint (e.g., `http://your-prometheus-instance:9090/api/v1/push`).
+- `solr_cluster_name`: Name of the Solr cluster.
+- `solr_host`: Hostname for Solr (e.g., `localhost`).
+- `solr_port`: Port for Solr (e.g., `9854`).
+- `solr_log_path`: Path to Solr log files (e.g., `/var/solr/logs/*.log`).
+- `instance_name`: Name of the Solr instance.
+- `loki_url`: URL for Loki push endpoint (e.g., `http://your-loki-instance:3100/loki/api/v1/push`).
+- `loki_user`: Your Loki username.
+- `loki_pass`: Your Loki password.
+
+# Debugging Tips
+
+When deploying the Apache Solr and ZooKeeper Cluster Sample App, you may encounter issues that require debugging. Here are some tips to validate the setup and troubleshoot common problems.
+
+## Accessing the Multipass Instances
+
+To access the Multipass instance, use the following command:
+```bash
+multipass shell apache-solr-zookeeper-instance-1
+```
+
+## Validating Services
```

Comment:
I went in the shell and don't know which services are not working. The PgBouncer does seem to be active when running `sudo systemctl status pgbouncer` I am not sure about the exporter. I am unable to get metrics back.